File: src/main/java/net/igneo/icv/entity/meteorSummoner/MeteorSummonerModel.java
package net.igneo.icv.entity.meteorSummoner;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class MeteorSummonerModel extends GeoModel<MeteorSummonerEntity> {
  private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
  private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
  private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");
  @Override
  public ResourceLocation getModelResource(MeteorSummonerEntity abyssStoneEntity) {
    return model;
  }
  
  @Override
  public ResourceLocation getTextureResource(MeteorSummonerEntity abyssStoneEntity) {
    return texture;
  }
  
  @Override
  public ResourceLocation getAnimationResource(MeteorSummonerEntity abyssStoneEntity) {
    return animations;
  }
}

File: src/main/java/net/igneo/icv/entity/chestplate/meteorSummoner/MeteorSummonerModel.kt
package net.igneo.icv.entity.chestplate.meteorSummoner

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class MeteorSummonerModel : GeoModel<MeteorSummonerEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")

    override fun getModelResource(abyssStoneEntity: MeteorSummonerEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(abyssStoneEntity: MeteorSummonerEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(abyssStoneEntity: MeteorSummonerEntity?): ResourceLocation {
        return animations
    }
}
